diff --git a/module.d.ts b/module.d.ts
index 36b86ffdadfb074a9fb7ba12adbfb954811d889d..fb530bae80d2c01cef2ff50384aaec15bdd4c7c9 100644
--- a/module.d.ts
+++ b/module.d.ts
@@ -249,7 +249,25 @@ declare module "module" {
              */
             lineLengths?: readonly number[] | undefined;
         }
-        interface SourceMapPayload {
+        /**
+         * https://tc39.es/source-map/#index-map
+         */
+        interface IndexSourceMapSection {
+            offset: {
+                line: number
+                column: number
+            }
+            map: BasicSourceMapPayload
+        }
+
+        // TODO(veil): Upstream types
+        /** https://tc39.es/ecma426/#sec-index-source-map */
+        interface IndexSourceMap {
+            version: number
+            file: string
+            sections: IndexSourceMapSection[]
+        }
+        interface BasicSourceMapPayload {
             file: string;
             version: number;
             sources: string[];
@@ -258,6 +276,7 @@ declare module "module" {
             mappings: string;
             sourceRoot: string;
         }
+        type SourceMapPayload = BasicSourceMapPayload | IndexSourceMap;
         interface SourceMapping {
             generatedLine: number;
             generatedColumn: number;
