diff --git a/module.d.ts b/module.d.ts
index 73b88af8b53babda5843dc5d4bb3b2a9b14d053b..6ecfa95f38c81421a49bc055bfff87a4f9dcc51c 100644
--- a/module.d.ts
+++ b/module.d.ts
@@ -49,7 +49,25 @@ declare module "module" {
          * @return Returns `module.SourceMap` if a source map is found, `undefined` otherwise.
          */
         function findSourceMap(path: string, error?: Error): SourceMap | undefined;
-        interface SourceMapPayload {
+        /**
+         * https://tc39.es/source-map/#index-map
+         */
+        interface IndexSourceMapSection {
+        offset: {
+            line: number
+            column: number
+        }
+        map: BasicSourceMapPayload
+        }
+
+        // TODO(veil): Upstream types
+        /** https://tc39.es/ecma426/#sec-index-source-map */
+        interface IndexSourceMap {
+        version: number
+        file: string
+        sections: IndexSourceMapSection[]
+        }
+        interface BasicSourceMapPayload {
             file: string;
             version: number;
             sources: string[];
@@ -58,6 +76,7 @@ declare module "module" {
             mappings: string;
             sourceRoot: string;
         }
+        type SourceMapPayload = BasicSourceMapPayload | IndexSourceMap;
         interface SourceMapping {
             generatedLine: number;
             generatedColumn: number;
