diff --git a/lib/Compilation.js b/lib/Compilation.js
index 66ae146f43914c45fc2892323b9977038bba9ff9..1fb370163773213a2eb1b78638d2610ba3f935f3 100644
--- a/lib/Compilation.js
+++ b/lib/Compilation.js
@@ -4621,7 +4621,10 @@ This prevents using hashes of each other and should be avoided.`);
 									}
 								} else if (!source) {
 									// render the asset
-									source = fileManifest.render();
+									const label = `asset render ${file}`
+									this.logger.time(label);
+ 									source = fileManifest.render();
+									this.logger.timeEnd(label);
 
 									// Ensure that source is a cached source to avoid additional cost because of repeated access
 									if (!(source instanceof CachedSource)) {